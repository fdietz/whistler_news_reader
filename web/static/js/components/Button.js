import React, { Component, PropTypes } from "react";

export default class Button extends Component {

  static PropTypes = {
    children: PropTypes.node.required,
    type: PropTypes.string.required,
    onClick: PropTypes.func
  };

  render() {
    const { type, onClick, children } = this.props;
    const cls = `btn ${type}`;

    return (
      <button onClick={onClick} className={cls}>
        {children}
      </button>
    );
  }
}
