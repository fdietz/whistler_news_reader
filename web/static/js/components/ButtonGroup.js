import React, { Component, PropTypes } from "react";

export default class ButtonGroup extends Component {

  static PropTypes = {
    children: PropTypes.node.isRequired
  };

  render() {
    return (
      <div className="btn-group btn-group-rounded">
        {this.props.children}
      </div>
    );
  }
}
