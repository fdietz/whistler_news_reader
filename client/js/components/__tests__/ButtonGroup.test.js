import renderer from 'react-test-renderer';
import React from 'react';
import ButtonGroup from '../ButtonGroup';
import Button from '../Button';

describe('ButtonGroup', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <ButtonGroup className="a">
        <Button type="primary">Hello</Button>
      </ButtonGroup>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
