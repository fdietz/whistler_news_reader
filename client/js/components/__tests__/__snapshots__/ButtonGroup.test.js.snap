exports[`ButtonGroup renders correctly 1`] = `
<div
  className="btn-group a">
  <button
    className="btn btn-primary"
    onClick={undefined}>
    Hello
  </button>
</div>
`;
